readthedocs/core/middleware.py

```diff
@@ -22,6 +22,11 @@
     'SINGLE_VERSION_URLCONF',
     'readthedocs.core.urls.single_version'
 )
+DEV_URL = getattr(
```

Comment:
`DEV_URL` might make more sense as `DEV_DOMAIN`, to match our other settings like `PRODUCTION_DOMAIN`. Optionally, `readthedocs.settings.dev` could just set `PUBLIC_DOMAIN` to this domain as well, unless we're using this for other logic.